docs/cudf/source/internals.md

```diff
@@ -0,0 +1,95 @@
+cuDF internals
+==============
+
+ColumnAccessor
+--------------
+
+The underlying data structure of cuDF `Series`, `DataFrame` and `Index` objects
+is a `ColumnAccessor`, which can be accessed via their `._data` attribute:
+
+```python
+>>> a = cudf.DataFrame({'x': [1, 2, 3], 'y': ['a', 'b', 'c']})
+>>> a._data
+ColumnAccessor(OrderedColumnDict([('x', <cudf.core.column.numerical.NumericalColumn object at 0x7f5a7d12e050>), ('y', <cudf.core.column.string.StringColumn object at 0x7f5a7d12e320>)]), multiindex=False, level_names=(None,))
+```
+
+`ColumnAccessor` is a dict-like object that maps column labels to columns.
+In addition, it supports things like selecting multiple columns (both by index and label),
+as well as hierarchical indexing.
```

Comment:
Would clarify that `ColumnAccessor` is ordered, as dict-like isn't indicative of that.